code/modules/tooltip/tooltip.dm

```diff
@@ -0,0 +1,103 @@
+/*
+Tooltips v1.1 - 22/10/15
+Developed by Wire (#goonstation on irc.synirc.net)
+- Added support for screen_loc pixel offsets. Should work. Maybe.
+- Added init function to more efficiently send base vars
+Configuration:
+- Set control to the correct skin element (remember to actually place the skin element)
+- Set file to the correct path for the .html file (remember to actually place the html file)
+- Attach the datum to the user client on login, e.g.
+	/client/New()
+		src.tooltips = new /datum/tooltip(src)
+Usage:
+- Define mouse event procs on your (probably HUD) object and simply call the show and hide procs respectively:
+	/obj/screen/hud
+		MouseEntered(location, control, params)
+			usr.client.tooltip.show(params, title = src.name, content = src.desc)
+		MouseExited()
+			usr.client.tooltip.hide()
+Customization:
+- Theming can be done by passing the theme var to show() and using css in the html file to change the look
+- For your convenience some pre-made themes are included
+Notes:
+- You may have noticed 90% of the work is done via javascript on the client. Gotta save those cycles man.
+- This is entirely untested in any other codebase besides goonstation so I have no idea if it will port nicely. Good luck!
+	- After testing and discussion (Wire, Remie, MrPerson, AnturK) ToolTips are ok and work for /tg/station13
+*/
+
+
+/datum/tooltip
+	var/client/owner
+	var/control = "mainwindow.tooltip"
+	var/file = 'code/modules/tooltip/tooltip.html'
+	var/showing = 0
```

Comment:
Right, so:

1. When a tooltip is shown for the first time ever, it is set to have showing = 1
2. If showing is 1, when hide() is called for the first time, queueHide is 0, so the hide code is called, and queueHide is set to 1
3. Then showing is set to 0, and because queueHide is set to 1, it calls hide() again, which runs the hide code, but behind a spawn for some reason
4. But if hide() was never called while showing was 1, hide() won't be called once showing is 0.

So why is the queueHide hide behind a spawn, and why would a queueHide hide be executed in the first place?